crates/gpui/src/platform/linux/wayland/window.rs

```diff
@@ -417,6 +417,66 @@ impl WaylandWindowStatePtr {
         }
     }
 
+    pub fn handle_ime_commit(&self, text: String) {
+        if let Some(ref mut fun) = self.callbacks.borrow_mut().input {
+            if !fun(PlatformInput::KeyUp(crate::KeyUpEvent {
+                keystroke: crate::Keystroke::default(),
```

Comment:
See `handle_key_event` in the macOS code for a rough example of the order of operations:

https://github.com/zed-industries/zed/blob/main/crates/gpui/src/platform/mac/window.rs#L1202-L1307

Reply:
Speaking of matching the macOS platform implementation: that one is in fact *horribly broken*, see #11829

Reply:
Yeah, that's a bad implementation of pre-edit. However, I believe the issue there lies with Zed, rather than with GPUI. I think it makes sense for GPUI to distribute both the raw keystrokes and the IME actions.

Reply:
Discussed this in discord, not possible to be similar to macOS with how wayland works, and this is good enough right now.